packages/ts-client/src/types.ts

```diff
@@ -11,34 +11,41 @@ import type {
 import type { AuthErrorReason } from './auth';
 import type { ReconnectConfig } from './reconnection';
 
-export type PeerServerMetadata<PeerMetadata> = {
-  peer: PeerMetadata;
-  server?: Record<string, unknown>;
+export type TrackMetadata = {
+  type: 'camera' | 'microphone' | 'screenShareVideo' | 'screenShareAudio';
+  paused: boolean;
+  // track label used in recordings
+  displayName?: string;
 };
 
-type TrackContextEvents<Metadata> = {
-  encodingChanged: (context: FishjamTrackContext<Metadata>) => void;
-  voiceActivityChanged: (context: FishjamTrackContext<Metadata>) => void;
+export type Metadata<P, S> = {
+  peer: P;
```

Comment:
Isn't peer metadata also optional?

Reply:
Thanks for noticing, actually neither of those should be.
The default type shall be `Record<string, unknown> | undefined`